spec/lib/jobs/uefi-job-spec.js

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+var uuid = require('node-uuid');
+
+describe('Run a UEFI Application', function () {
+    var UefiJob;
+    var subscribeRequestProfileStub;
+    var subscribeRequestPropertiesStub;
+    var subscribeHttpResponseStub;
+    var job;
+
+    before(function () {
+        helper.setupInjector(
+            _.flattenDeep([
+                helper.require('/lib/jobs/base-job'),
+                helper.require('/lib/jobs/uefi-job')
+            ])
+        );
+        UefiJob = helper.injector.get('Job.Uefi');
+        subscribeRequestProfileStub = sinon.stub(
+            UefiJob.prototype, '_subscribeRequestProfile');
+        subscribeRequestPropertiesStub = sinon.stub(
+            UefiJob.prototype, '_subscribeRequestProperties');
+        subscribeHttpResponseStub = sinon.stub(
+            UefiJob.prototype, '_subscribeHttpResponse');
+    });
+
+    beforeEach(function () {
+        subscribeRequestProfileStub.reset();
+        subscribeRequestPropertiesStub.reset();
+        subscribeHttpResponseStub.reset();
+        job = new UefiJob(
+            {
+                profile: 'testprofile',
+                repo: 'http://127.0.0.1:8080',
+                uefitool: 'testuefi.efi',
+                args: 'args'
+            },
+            {
+                target: 'testid'
+            },
+             uuid.v4());
+    });
+
+    after(function () {
+        subscribeRequestProfileStub.restore();
+        subscribeRequestPropertiesStub.restore();
+        subscribeHttpResponseStub.restore();
+    });
+
+    it("should have a nodeId value", function () {
+        expect(job.nodeId).to.equal('testid');
+    });
+
+    it("should have a profile value", function () {
+        expect(job.profile).to.equal('testprofile');
+    });
+
+    it("should set up message subscribers", function () {
+        var cb;
+        return job._run().then(function () {
```

Comment:
There doesn't really seem to be a reason to have the `_run()` method return a promise at all. In test, you could just call `job._run()` and then check your stubs, since it will all be called synchronously.